src/formatter-settings/assets/features/formatterSettings/components/InsertLine.tsx

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import React from "react";
+import { useSelector} from "react-redux";
+import { Catagory, JavaFormatterSetting } from "../../../../types";
+import Setting from "./Setting";
+
+const InsertLine = (): JSX.Element => {
+  const settings: JavaFormatterSetting[] = useSelector((state: any) => state.formatterSettings.settings);
+
+  return (
+    <Setting setting={settings} catagory={Catagory.InsertLine}/>
```

Comment:
You are already using redux, is it necessary still passing props?

Reply:
It's unnecessary, and we can remove all the category components and use activeCategory in slice's state instead.